libs/community/langchain_community/utilities/atlassian.py

```diff
@@ -0,0 +1,357 @@
+"""Util that calls Atlassian APIs."""
+
+import inspect
+import json
+import re
+from typing import Any, List, Optional
+
+from langchain_core.utils import get_from_dict_or_env
+from pydantic import BaseModel, ConfigDict
+
+
+class AtlassianAPIWrapper(BaseModel):
+    """Wrapper for Atlassian APIs, including Jira and Confluence."""
+
+    jira: Optional[Any] = None
+    confluence: Optional[Any] = None
+    atlassian_username: Optional[str] = None
+    atlassian_api_token: Optional[str] = None
+    atlassian_instance_url: Optional[str] = None
+    atlassian_cloud: Optional[bool] = None
+    filter_keys: Optional[List[str]] = None
+
+    model_config = ConfigDict(extra="forbid")
+
+    def __init__(self, **data):
+        super().__init__(**data)
+        self.validate_and_initialize_environment()
+
+    def validate_and_initialize_environment(self):
+        """Validates if the API key and Python package are present in the environment
+        and initializes the Jira and Confluence instances."""
+
+        def get_env_var(
+            var_name: str, env_name: str, default: Optional[str] = None
+        ) -> str:
+            return get_from_dict_or_env(self.__dict__, var_name, env_name, default)
+
+        self.atlassian_username = get_env_var(
+            "atlassian_username", "ATLASSIAN_USERNAME", ""
+        )
+        self.atlassian_api_token = get_env_var(
+            "atlassian_api_token", "ATLASSIAN_API_TOKEN"
+        )
+        self.atlassian_instance_url = get_env_var(
+            "atlassian_instance_url", "ATLASSIAN_INSTANCE_URL"
+        )
+        self.atlassian_cloud = bool(get_env_var("atlassian_cloud", "ATLASSIAN_CLOUD"))
```

Comment:
How to handle if jira and confluence are not hosted under the same url. Could we have separate variable for both jira and confluence ? I try with the current master branch, and I need to declare OS env variable JIRA_USERNAME, JIRA_API_TOKEN, JIRA_INSTANCE_URL to get it to work. Passing the variables in the JiraAPIWrapper does not handle it right (causing errors in the agent workflow). So I can't declare on the same host Jira and Confluence because it uses the same env variable. thanks

Reply:
@jeffWild Thank you for your feedback! It makes perfect sense to have separate variables for Jira and Confluence, especially since it's common to host them separately in self-hosted environments. I will work on this improvement to allow for distinct environment variables for each service (like CONFLUENCE_USERNAME, CONFLUENCE_API_TOKEN, CONFLUENCE_INSTANCE_URL), and I'll publish an update soon.

Thanks for pointing this out!